// react-native-code-highlighter/dist/commonjs/utils/styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getRNStylesFromHljsStyle = exports.cleanStyle = void 0;
var _cssToReactNative = _interopRequireDefault(require("css-to-react-native"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const ALLOWED_STYLE_PROPERTIES = {
  color: true,
  background: true,
  backgroundColor: true,
  fontWeight: true,
  fontStyle: true
};
const getRNStylesFromHljsStyle = hljsStyle => {
  return Object.fromEntries(Object.entries(hljsStyle).map(_ref => {
    let [className, style] = _ref;
    return [className, cleanStyle(style)];
  }));
};
exports.getRNStylesFromHljsStyle = getRNStylesFromHljsStyle;
const cleanStyle = style => {
  const styles = Object.entries(style).filter(_ref2 => {
    let [key] = _ref2;
    return ALLOWED_STYLE_PROPERTIES[key];
  }).map(_ref3 => {
    let [key, value] = _ref3;
    return [key, value];
  });
  return (0, _cssToReactNative.default)(styles);
};
exports.cleanStyle = cleanStyle;
